### ISSUE TITLE
Inconsistency between API call responses

### ISSUE THREAD

**Issue by saxtouri** [eu-digital-green-certificates/dgca-businessrule-service#30]
The following problem was reported on the internal demo deployment of the Greek infrastructure.

## Describe the bug
 API call 1: GET /valuesets --> returns a list of (id, hash)
 API call 2: GET /valuesets/{hash} --> returns info about a valueset.
 
The hash for the "covid-19-lab-test-manufacturer-and-name" valueset is returned in API call 1, but when we query for its has wth API call 2, we get this response:
```
{
  "code": "0x001",
  "problem": "Possible reasons: The provided hash value is not correct",
  "sendValue": "<sniped>",
  "details": ""
}
```

## Expected behaviour

Every other hash in the list returned by API call 1, also produces a good response from API call 2,

## Technical details

We have deployed the provided container image and deployed on k8s.

Can you guide us to fix the problem? Is it a bug or a misconfiguration?

**Comment by psavva** [comment]
@saxtouri , If you where using Swagger for your testing, make sure you didn't use one of the hashes which are presented as part of the swagger examples.

I am using this with no issues what so ever, I highly doubt this is a valid bug.

**Comment by saxtouri** [comment]
Well, I'm not talking about a local test, I'm talking about the "official" demo deployment. Since we use the published containers to run this deployment (not the source code) but I don't think we/they used any sample hashes from the swagger. Still, I will relay your suggestion to the deployment team.

In the meantime, if this is not a bug, can you suggest how we can explore and, ultimately, solve the issue? We need some help.

**Comment by slaurenz** [comment]
@saxtouri,
in the example above it seams, that the hash value have been replaced with "<sniped>". For investigating the problem, it would be good to see the real hash code. Then you can check the DB, if a value set exist with that hash code. For more advise it would be good to have some more Information about the setup, in general to which gateway the service is connected and what profiles are used.
regards
Simon

**Comment by louridas** [comment]
> @saxtouri, in the example above it seams, that the hash value have been replaced with "". For investigating the problem, it would be good to see the real hash code. Then you can check the DB, if a value set exist with that hash code. For more advise it would be good to have some more Information about the setup, in general to which gateway the service is connected and what profiles are used. regards Simon

@slaurenz  You can check the situation for yourself:

```
curl 'https://businessrule.dgc.demo.gov.gr/valuesets'
[{"id":"country-2-codes","hash":"bb1bc3b46cfc9913dceef8c362675eb810a42babb96fb23bee02275d7bdfbd4c"},
{"id":"covid-19-lab-result","hash":"64f946691cc68966335da6dfe16d4177de8c5d3ce6abc2cf18c30103a70763d6"},
{"id":"covid-19-lab-test-manufacturer-and-name","hash":"15b9c0682be6658bcfc40ed322239b3308bba2b76cce68075ff81fbab492edd5"},
{"id":"covid-19-lab-test-type","hash":"a414db002259278577e5e29241096936ac955b27c0f09ebd6c1a84ca8334ee1c"},
{"id":"disease-agent-targeted","hash":"50c3e3bd108e955e209ff428ef5ce4751514fef892d9ed8912b89a1f1ce89559"},
{"id":"sct-vaccines-covid-19","hash":"c1f0b3585687f51eb510267b5fc5d4471f2289eb3d43d547c493e0ae2ea498b7"},
{"id":"vaccines-covid-19-auth-holders","hash":"00c05eac57d98379601309761b7989c92931df778758fec9692b577475dd2c21"},
{"id":"vaccines-covid-19-names","hash":"2785a359333ae9370fb2ef89c05f1d88aedcde3b8b4125c267a891046add72d7"}]
```

and then trying to get the hashvalue `15b9c0682be6658bcfc40ed322239b3308bba2b76cce68075ff81fbab492edd5`:

```
curl 'https://businessrule.dgc.demo.gov.gr/valuesets/15b9c0682be6658bcfc40ed322239b3308bba2b76cce68075ff81fbab492edd5'
{"code":"0x001","problem":"Possible reasons: The provided hash value is not correct","sendValue":"15b9c0682be6658bcfc40ed322239b3308bba2b76cce68075ff81fbab492edd5","details":""}
```

**Comment by slaurenz** [comment]
Hi @saxtouri,
The data for both requests should be based on the VALUESETS table in the DB. 
If the issue still occures, can you please provide some more Information:

From the DB:
- SELECT * FROM VALUESETS;
- SELECT * FROM SIGNED_LIST;

From the API:
- The response from the /valuesets request

General:
Setup of the environment like number of pods 

Regards Simon

**Comment by saxtouri** [comment]
Hi @slaurenz 

The demo deployment consists of three containers, one for each image:
```console
$ kubectl -n green-pass get pods
NAME                                READY   STATUS    RESTARTS   AGE
dgc-businessrule-6789abcd12-ef345   1/1     Running   0          15d
dgc-crl-67g8h9ik12-345lm            1/1     Running   0          15d
dgc-verifier-67891n2345-6o7pq       1/1     Running   0          15d
```

From the DB cli we get:
```
dgc_businessrule=> SELECT * FROM VALUESETS;
                               hash                               |             identifier_name             | raw_data 
------------------------------------------------------------------+-----------------------------------------+----------
 64f946691cc68966335da6dfe16d4177de8c5d3ce6abc2cf18c30103a70763d6 | covid-19-lab-result                     | 54519
 50c3e3bd108e955e209ff428ef5ce4751514fef892d9ed8912b89a1f1ce89559 | disease-agent-targeted                  | 54520
 c1f0b3585687f51eb510267b5fc5d4471f2289eb3d43d547c493e0ae2ea498b7 | sct-vaccines-covid-19                   | 54521
 a414db002259278577e5e29241096936ac955b27c0f09ebd6c1a84ca8334ee1c | covid-19-lab-test-type                  | 54523
 269b0b976878a4585a95ea1d8a877c74540a8cfc4e9113fbe41c8514ed441a18 | vaccines-covid-19-auth-holders          | 54835
 adbdaf4cd18c4d281fbd165c47ea8f83fb5ca9b0f6298cd695ea550f663ea524 | vaccines-covid-19-names                 | 54836
 669ef04a4a5676597093f18b7b01df9d1498003e312e37a09758a84ea28f5c9d | country-2-codes                         | 54838
 e0efdb1b68295f9ac096c418d11c0e0a14c06a7667566b0501f79a84010bc170 | covid-19-lab-test-manufacturer-and-name | 55060
(8 rows)  

dgc_businessrule=> SELECT * FROM SIGNED_LIST;
 list_type |                               hash                               | raw_data | signature 
-----------+------------------------------------------------------------------+----------+-----------
 Rules     | 092aa19580b39e6f922a65d5b9f6ecb2ac731d652f49c6b31e983e6dfb9c40c7 | 55080    | 
 ValueSets | d16508467bee8325075e55855da8cfb229ac58f3536a66ee56bab61ae834ffcd | 55061    | 
(2 rows)
```

and this is the API response:
```json
$ curl https://businessrule.dgc.demo.gov.gr/valuesets
[
    {
        "hash": "bb1bc3b46cfc9913dceef8c362675eb810a42babb96fb23bee02275d7bdfbd4c",
        "id": "country-2-codes"
    },
    {
        "hash": "64f946691cc68966335da6dfe16d4177de8c5d3ce6abc2cf18c30103a70763d6",
        "id": "covid-19-lab-result"
    },
    {
        "hash": "f87b3419504999fdd4083dca3a816a2ea69af8c435718df9da7ef4372dc61ae7",
        "id": "covid-19-lab-test-manufacturer-and-name"
    },
    {
        "hash": "a414db002259278577e5e29241096936ac955b27c0f09ebd6c1a84ca8334ee1c",
        "id": "covid-19-lab-test-type"
    },
    {
        "hash": "50c3e3bd108e955e209ff428ef5ce4751514fef892d9ed8912b89a1f1ce89559",
        "id": "disease-agent-targeted"
    },
    {
        "hash": "c1f0b3585687f51eb510267b5fc5d4471f2289eb3d43d547c493e0ae2ea498b7",
        "id": "sct-vaccines-covid-19"
    },
    {
        "hash": "00c05eac57d98379601309761b7989c92931df778758fec9692b577475dd2c21",
        "id": "vaccines-covid-19-auth-holders"
    },
    {
        "hash": "2785a359333ae9370fb2ef89c05f1d88aedcde3b8b4125c267a891046add72d7",
        "id": "vaccines-covid-19-names"
    }
]
```

**Comment by saxtouri** [comment]
Hi @slaurenz,

our container image is build on commit #181d9fa which was the 1.1.1 release at the time of deployment. We discovered that there is a commit promising to fix our issue (#8652aa2 "fix list update"). We are going to try the corresponding container ( https://github.com/eu-digital-green-certificates/dgca-businessrule-service/pkgs/container/dgca-businessrule-service%2Fdgca-businessrule-service/6125226?tag=1.1.2 )

Nevertheless, we are confused regarding the release strategy of the package. We can see is 1.1.2 acts as a patch release for 1.1.1.

Is there a way to be alerted about important bug or security fixes?

We will let you know if the update to 1.1.2 fixed the problem.

**Comment by saxtouri** [comment]
OK, deploying the 1.1.2 container fixed the issue. I would appreciate it if the ticket remains open for a few more days until we are certain, but our tests show that the issue is resolved.

What surprises me is that the bug I reported is an actual bug which was spotted and fixed. When we deployed the application, the bug was in the code of the latest master release. When we reported the bug, the fix commit was already in the repository, but the 1.1.1 release was not updated with it.

We are responsible for failing to provide release and deployment information when I first reported the bug. That would have help resolve the issue faster. Still, it is surprising how no one from the development team was able to recall this as an actual bug you had already fixed on a later release.

I guess, both sides need to revisit versioning and release issues.
